gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -626,38 +615,36 @@ private String computeWatermarkText() {
       String nameAuthor = "";
       String nameForWatermark = "";
       Photo photo = getMedia();
-      if (watermark && photo.getFileMimeType().isIPTCCompliant()) {
-        try {
-          if (isDefined(watermarkHD)) {
-            nameAuthor = processWatermarkHD(nameAuthor, photo);
-          }
-          if (isDefined(watermarkOther)) {
-            nameAuthor = defaultStringIfNotDefined(getWatermarkValue(watermarkOther), nameAuthor);
-            if (!nameAuthor.isEmpty()) {
-              nameForWatermark = nameAuthor;
+      if (watermark != null && watermark.isEnabled()) {
+          try {
+            nameAuthor = processWatermarkHD(photo);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "nameAuthor". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1854)